app/components/Form/ProjectManagerForm.tsx

```diff
@@ -38,27 +69,168 @@ const ProjecManagerForm: React.ForwardRefRenderFunction<any, Props> = (
         }
       }
     `,
-    props.allUsers
+    props.query
   );
 
-  const schema: JSONSchema7 = useMemo(() => {
-    const initialSchema = projectManagerSchema;
-
-    initialSchema.properties.cifUserId = {
-      ...initialSchema.properties.cifUserId,
+  // Dynamically build the schema from the list of cif_users
+  const managerSchema = useMemo(() => {
+    const schema = projectManagerSchema;
+    schema.properties.cifUserId = {
+      ...schema.properties.cifUserId,
       anyOf: allCifUsers.edges.map(({ node }) => {
         return {
           type: "number",
           title: `${node.firstName} ${node.lastName}`,
           enum: [node.rowId],
           value: node.rowId,
-        };
+        } as JSONSchema7Definition;
       }),
     };
-    return initialSchema as JSONSchema7;
+
+    return schema as JSONSchema7;
   }, [allCifUsers]);
 
-  return <FormBase {...props} ref={ref} schema={schema} uiSchema={uiSchema} />;
+  // Add a manager to the project revision
+  const [applyAddManagerToRevision] = useAddManagerToRevisionMutation();
+  const addManager = (data: any) => {
+    applyAddManagerToRevision({
+      variables: {
+        projectRevision: projectRevision.id,
+        projectRevisionId: projectRevision.rowId,
+        newFormData: data,
+      },
```

Comment:
can we have an optimistic response here? The component flashes with an empty value and then the value changes when the network request succeeds